assets/src/edit-story/components/canvas/pagemenu/pageMenu.js

```diff
@@ -0,0 +1,186 @@
+  width: var(--pagemenu-icon-space);
+`;
+
+function PageMenu() {
+  const {
+    state: { canUndo, canRedo },
+    actions: { undo, redo },
+  } = useHistory();
+  const {
+    currentPageNumber,
+    currentPage,
+    deleteCurrentPage,
+    addPage,
+    hasAnimations,
+  } = useStory(
+    ({
+      state: { currentPageNumber, currentPage },
+      actions: { deleteCurrentPage, addPage },
+    }) => {
+      return {
+        currentPageNumber,
+        currentPage,
+        deleteCurrentPage,
+        addPage,
+        hasAnimations: currentPage?.animations?.length > 0,
+      };
+    }
+  );
+  const { pageSize } = useCanvas((state) => ({
+    pageSize: state.state.pageSize,
+  }));
+  const { isRTL } = useConfig();
+
+  const handleDeletePage = useCallback(() => deleteCurrentPage(), [
+    deleteCurrentPage,
+  ]);
+
+  const handleAddPage = useCallback(() => addPage({ page: createPage() }), [
+    addPage,
+  ]);
+
+  const handleDuplicatePage = useCallback(
+    () => addPage({ page: duplicatePage(currentPage) }),
+    [addPage, currentPage]
+  );
+
+  const isWidePage = pageSize.width > 280;
+
+  const handleUndo = useCallback(() => undo(), [undo]);
+
+  const handleRedo = useCallback(() => redo(), [redo]);
+
+  if (!currentPage) {
+    return null;
+  }
+
+  return (
+    <Wrapper isWidePage={isWidePage}>
+      <PageCount>
+        {sprintf(
+          /* translators: %s: page number. */
+          __('Page %s', 'web-stories'),
+          currentPageNumber
+        )}
+      </PageCount>
+      <CountSpace />
+      <PageMenuButton
+        title={__('Delete page', 'web-stories')}
+        onClick={handleDeletePage}
+        aria-label={__('Delete Page', 'web-stories')}
+      >
+        <Icons.Trash />
+      </PageMenuButton>
+      <IconSpace />
+      <PageMenuButton
+        title={__('Duplicate page', 'web-stories')}
+        onClick={handleDuplicatePage}
+        aria-label={__('Duplicate Page', 'web-stories')}
+      >
+        <Icons.PagePlus />
+      </PageMenuButton>
+      <IconSpace />
+      <PageMenuButton
+        title={__('New page', 'web-stories')}
+        onClick={handleAddPage}
+        aria-label={__('Add New Page', 'web-stories')}
+      >
+        <Icons.PlusOutline />
+      </PageMenuButton>
+      <IconSpace />
+      <Divider />
+      <IconSpace />
+      <PageMenuButton
+        title={__('Undo', 'web-stories')}
+        shortcut="mod+z"
+        disabled={!canUndo}
+        onClick={handleUndo}
+        aria-label={__('Undo Changes', 'web-stories')}
+      >
+        {isRTL ? <Icons.ArrowDownrightCurved /> : <Icons.ArrowDownleftCurved />}
+      </PageMenuButton>
+      <IconSpace />
+      <PageMenuButton
+        title={__('Redo', 'web-stories')}
+        shortcut="shift+mod+z"
+        disabled={!canRedo}
+        onClick={handleRedo}
+        aria-label={__('Redo Changes', 'web-stories')}
+      >
+        {isRTL ? <Icons.ArrowDownleftCurved /> : <Icons.ArrowDownrightCurved />}
```

Comment:
Nit: Should `left` and `right` be with capital letters, too? `ArrowDownLeftCurved` etc.

Reply:
Done in [the icon PR](https://github.com/google/web-stories-wp/pull/6311/commits/4b24d3ce8c32d41a1c6daac2ef80bd024c8ea67d).